### DISCUSSION TITLE
moving panels

### DISCUSSION THREAD

**Discussion by ctfrancia** [JetBrains/ideavim#728] [Q&A]
So I am trying out the IDE and I'm coming from VIM, I understand that in the root I can put my remappings in the `.ideavimrc` which I have and for the most part working great. However, what I am trying to do is to move panels without having to use my mouse. So for example I am trying to move from the right panel (where one writes code) to the tree viewer on the left. Currently in vim I am using NERDTree and have in my `init.vim`
```
" maps split navigation
nnoremap <C-J> <C-W><C-J>
nnoremap <C-K> <C-W><C-K>
nnoremap <C-L> <C-W><C-L>
nnoremap <C-H> <C-W><C-H>
```

the above remappings will allows me to hold control while more pressing L or K or J or H and move around various window panes. The same snippet above placed within my `.ideavimrc` doesn't allow me to change window pane focus.

**Comment by citizenmatt** [comment]
Unfortunately, this isn't possible right now. IdeaVim is mainly an extension to the IDE's editor, and so most of the functionality only works in the editor (with the exception of NERDTree shortcuts in the Project view). IdeaVim key mappings only work in the editor, and window movement only moves between editor windows. Your best option right now is to use the default shortcut for opening the Project view, or create a mapping such as `nmap <leader>p <Action>(ActivateProjectToolWindow)`.

**Comment by Ste-Mich** [comment]
You can add this to .ideavimrc:

nnoremap <C-h> <C-w>h
nnoremap <C-j> <C-w>j
nnoremap <C-k> <C-w>k
nnoremap <C-l> <C-w>l

This allows you to navigate out of the code window, but not back into it.

For NERDTree you can install it in Ideavim, though i haven't tried it yet.

In .ideavimrc:

Plug 'preservim/nerdtree'
